### PR TITLE
Move  sql connection factory.create connection to base repository

### PR DESCRIPTION
**Pull request by Gredja**

### CODE REVIEW

File: src/Softeq.NetKit.Chat.Data.Persistent/Database/IDatabaseManager.cs [hunk 1]
```diff
@@ -8,6 +8,6 @@ namespace Softeq.NetKit.Chat.Data.Persistent.Database
     public interface IDatabaseManager
     {
         Task CreateEmptyDatabaseIfNotExistsAsync();
-        void MigrateToLatestVersion();
+        Task MigrateToLatestVersion();
```
Comment (eugepae) [inline src/Softeq.NetKit.Chat.Data.Persistent/Database/IDatabaseManager.cs]: Async

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Database/DatabaseManager.cs [hunk 2]
```diff
@@ -43,11 +43,11 @@ public async Task CreateEmptyDatabaseIfNotExistsAsync()
             }
         }
 
-        public void MigrateToLatestVersion()
+        public async Task MigrateToLatestVersion()
```
Comment (eugepae) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Database/DatabaseManager.cs]: Async

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/BaseRepository.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using EnsureThat;
+using Softeq.NetKit.Chat.Data.Persistent.Sql.Database;
+
+namespace Softeq.NetKit.Chat.Data.Persistent.Sql.Repositories
+{
+    public abstract class BaseRepository
```
Comment (eugepae) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/BaseRepository.cs]: Do we really need abstract class?

File: src/Softeq.NetKit.Chat.Data.Persistent/Database/IDatabaseManager.cs (hunk 1 above)
Comment (Gredja) [reply src/Softeq.NetKit.Chat.Data.Persistent/Database/IDatabaseManager.cs]: Done

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/BaseRepository.cs (hunk 3 above)
Comment (Gredja) [reply src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/BaseRepository.cs]: Done

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Database/DatabaseManager.cs (hunk 2 above)
Comment (Gredja) [reply src/Softeq.NetKit.Chat.Data.Persistent.Sql/Database/DatabaseManager.cs]: Done

**Review by eugepae — APPROVED** [review]

**Review by eugenekirvel — APPROVED** [review]

**Pull request merged**